test/cqlpy/conftest.py

```diff
@@ -41,13 +41,20 @@ def pytest_addoption(parser):
     add_s3_options(parser)
 
 
+@pytest.fixture(scope="module")
+async def host(request, testpy_test: PythonTest):
+    if request.config.getoption("--test-py-init"):
+        async with testpy_test.run_ctx(options=testpy_test.suite.options):
+            yield testpy_test.server_address
+    else:
+        yield request.config.getoption("--host")
```

Comment:
Wow, now I'm even more worried - I just realized that not only that for Alternator tests you will now create and delete test tables 40 times (once for each file), you'll also boot and shut down Scylla 40 times. Is this true? What a waste that will be.

But besides that, I think this extra host fixture is a mistake. These new lines of code could have been part of the existing "cql" fixture. This would not only minimize the churn, there would also be important for allowing the "cql" fixture (and the Scylla node connected to it) to live throughout the life of the session, in parallel to a "alternator" fixture (for example), which will need a **different** host to be created. You can't have just one "host" fixture if there are multiple ways of creating that host.

Reply:
@nyh - given the above comment, what else is needed here? Can we resolve this conversation?